go.mod

```diff
@@ -1,13 +1,13 @@
 module github.com/honeycombio/honeycomb-opentelemetry-go
 
-go 1.20
+go 1.21
 
 require (
 	github.com/honeycombio/otel-config-go v1.13.1
 	github.com/stretchr/testify v1.9.0
-	go.opentelemetry.io/otel v1.24.0
-	go.opentelemetry.io/otel/exporters/stdout/stdouttrace v1.23.0
-	go.opentelemetry.io/otel/sdk v1.23.0
+	go.opentelemetry.io/otel v1.25.0
```

Comment:
Should we wait and update otel-config-go once it's released?

Reply:
I think it's fine to let this change happen and update otel-config-go in a separate PR. The dependency resolver will figure out the min otel version between direct & indirect deps.

Reply:
Updated otel-config-go as well